ddtrace/contrib/asyncpg/patch.py

```diff
@@ -0,0 +1,122 @@
+from typing import TYPE_CHECKING
+
+from ddtrace import Pin
+from ddtrace import config
+from ...constants import SPAN_MEASURED_KEY
+
+from ...ext import SpanTypes
+from ...ext import db
+from ...ext import net
+from ...internal.utils import get_argument_value
+from ..trace_utils import ext_service
+from ..trace_utils import unwrap
+from ..trace_utils import with_traced_module as with_traced_module_sync
+from ..trace_utils import wrap
+from ..trace_utils_async import with_traced_module
+
+
+if TYPE_CHECKING:
+    from types import ModuleType
+    from typing import Dict
+
+    import asyncpg
+
+config._add(
+    "asyncpg",
+    dict(
+        _default_service="postgres",
+    ),
+)
+
+
+def _get_connection_tags(conn):
+    # type: (asyncpg.Connection) -> Dict[str, str]
+    addr = conn._addr
+    params = conn._params
+    host = port = ""
+    if isinstance(addr, tuple) and len(addr) == 2:
+        host, port = addr
+    return {
+        net.TARGET_HOST: host,
+        net.TARGET_PORT: port,
+        db.USER: params.user,
+        db.NAME: params.database,
+    }
+
+
+@with_traced_module
+async def _traced_connect(asyncpg, pin, func, instance, args, kwargs):
+    with pin.tracer.trace(
+        "postgres.connect", span_type=SpanTypes.SQL, service=ext_service(pin, config.asyncpg)
+    ) as span:
+        conn = await func(*args, **kwargs)
+        conn_pin = pin.clone()
+        conn_pin.onto(conn)  # TODO: connection uses slots
```

Comment:
probably need to create an ObjectProxy wrapper for the connection, similar to psycopg2/etc